forest/daemon/src/main.rs

```diff
@@ -190,7 +197,21 @@ fn main() {
                     }
 
                     let rt = Runtime::new().unwrap();
-                    rt.block_on(daemon::start(cfg, opts.detach));
+                    let db: Db = rt.block_on(daemon::start(cfg, opts.detach));
+
+                    info!("Shutting down tokio...");
+                    rt.shutdown_timeout(Duration::from_secs(10));
+
+                    let db_weak_ref = Arc::downgrade(&db.db);
+                    drop(db);
+
+                    if db_weak_ref.strong_count() != 0 {
+                        error!(
+                            "Dangling reference to DB detected: {}. Tracking issue: https://github.com/ChainSafe/forest/issues/1891",
+                            db_weak_ref.strong_count()
+                        );
+                    }
```

Comment:
I would be fine with not having the dangling DB references check anymore. Or we could leave it in for now and then remove it later once we're 100% confident it can't happen anymore.